src/lib/yang/snabb-softwire-v1.yang

```diff
@@ -0,0 +1,381 @@
+        identifier\" (PSID) from a given port.
+
+                             0                   1
+                             0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5
+                            +-----------+-----------+-------+
+              Ports in      |     A     |    PSID   |   j   |
+           the CE port set  |    > 0    |           |       |
+                            +-----------+-----------+-------+
+                            |  a bits   |  k bits   |m bits |
+
+             Figure 2: Structure of a Port-Restricted Port Field
+
+          Source: http://tools.ietf.org/html/rfc7597#section-5.1
+
+        We find the specification's names to be a bit obtuse, so we
+        refer to them using the following names:
+
+          a bits = reserved-ports-bit-count.
+          k bits = psid-length.
+          m bits = shift.";
+
+      key addr;
+
+      leaf addr {
+        type inet:ipv4-address;
+        mandatory true;
+        description
+         "Public IPv4 address managed by the lwAFTR.";
+      }
+
+      leaf end-addr {
+        type inet:ipv4-address;
+        description
+         "If present, this PSID map entry applies to all addresses
+          between 'addr' and this address, inclusive.";
+      }
+
+      leaf psid-length {
+        type uint8 { range 0..16; }
+        mandatory true;
+        description
+         "The number of bits devoted to the PSID in the port map.
+          If the psid-length is N, then the IPv4 address will be
+          shared 2^N ways.  Note that psid-length, shift, and
+          reserved-ports-bit-count must add up to 16.";
+      }
+
+      leaf shift {
+        type uint8 { range 0..16; }
+        description
+         "Given an incoming port, one can obtain the PSID by
+          shifting the port right by 'shift' bits and then masking
+          off the lowest 'psid-length' bits.  Defaults to 16 -
+          psid-length.  Note that psid-length, shift, and
+          reserved-ports-bit-count must add up to 16.";
+      }
+
+      leaf reserved-ports-bit-count {
+        type uint8 { range 0..16; }
+        default 0;
+        description
+         "Reserve the lowest 2^N ports so that they map to no
+          softwire.  This can be useful to prevent the low 1024
+          ports (for example) from being mapped to customers.  Note
+          that psid-length and shift must add up to less than or
+          equal to 16.";
+      }
+    }
+
+    leaf-list br-address {
+      type inet:ipv6-address;
+      description
+       "B4-facing address of an lwAFTR.";
+    }
+
+    list softwire {
+      key "ipv4 psid padding";
+
+      leaf ipv4 {
+        type inet:ipv4-address;
+        mandatory true;
+        description
+         "Public IPv4 address of the softwire.";
+      }
+
+      leaf psid {
+        type uint16;
+        mandatory true;
+        description
+         "Port set ID.";
+      }
+
+      leaf padding {
+        type uint16 { range 0..0; }
+        default 0;
```

Comment:
I think i'm wrong in suggesting it's not permitted, it is, it's just ignored as the RFC says. I think the best situation here is to remove it to avoid confusion, this value has to be provide even though we explicitly don't want them to... (hence why we've put a single digit range and default.